src/nnet3bin/nnet3-xvector-get-egs-seg.cc

```diff
@@ -0,0 +1,166 @@
+// nnet3bin/nnet3-xvector-get-egs-seg.cc
+
+// Copyright 2016-2017  Johns Hopkins University (author:  Hang Lyu)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include <sstream>
+#include "util/common-utils.h"
+#include "nnet3/nnet-example.h"
+
+namespace kaldi {
+namespace nnet3 {
+
+// Process the utt2label file and store it as a map from utt_id to
+// label
+static void ProcessUttToLabelFile(const std::string utt2label_rxfilename, 
+    unordered_map<std::string, int32> *utt_to_label) {
+  Input utt2label_input(utt2label_rxfilename);
+  if (!utt2label_rxfilename.empty()) {
+    std::string line;
+    while (std::getline(utt2label_input.Stream(), line)) {
+      std::vector<std::string> fields;
+      SplitStringToVector(line, " \t\n\r", true, &fields);
+      if (fields.size() != 2) {
+        KALDI_ERR << "Expected 2 fields in line of utt2label file, got "
+                  << fields.size() << " instead.";
+      }
+      std::string utt_id = fields[0];
+      int32 label_id;
+      if (!ConvertStringToInteger(fields[1], &label_id)) {
+        KALDI_ERR << "Expect integer for label_id";
+      }
+      // Add to map
+      (*utt_to_label)[utt_id] = label_id;
+    } // end of while 
+  } // end of if
+}
+
+
+static void WriteExamples(const MatrixBase<BaseFloat> &feats,
+  const int32 &this_pdf_id, const std::string &key,
+  bool compress, int32 num_pdfs, int32 &num_egs_written,
+  NnetExampleWriter &example_writer) {
```

Comment:
The Google standard (https://google.github.io/styleguide/cppguide.html#Reference_Arguments) is to use pointers for "output" variables instead of references.